feat(contact message): contact message queries

Shumba250 opened the pull request:
created API endpoint of contact messages 

. created GET API that retrieves all messages from the database
. created GET API that retrieves a single message from the database
. created GET API that retrieves the number of messages from the database
. created POST API that sends a message to the database
.created DELETE API that deletes a message

Shumba250 committed:
connected to server at port 3000, connected the server to mongoDB and tested GET, POST,PATCH, DELETE  endpoints and added create single post, read all posts, update the blog and delete a post

Shumba250 committed:
ch(messages): contact message queries
created GET API to retrieve all messages
created GET API to retireve a single message
created POST API to post a message to database
created DELETE API to delete a single message
all endpoints tested successfully with postman

review contactquery

wakaflorien commented:
> 
good PR but can be improved.

### What does this PR do?
[what the PR does]
### Description of Task to be completed
[the tasks that you completed on the feature]
### How should this be manually tested? 
### Any background context you want to provide? (if necessary)

@Shumba250

wakaflorien commented on server/routes/router.js:
```diff
@@ -0,0 +1,65 @@
+			image: req.body.image,
+		});
+		const blogs = await blog.save();
+		res.status(200).json(blogs);
```
Looks good but i would like responses like this 

`res.status(200).json({"status":"success", "message":"all posts", "data":{"posts":posts}})`
`res.status(404).json({ "status":"success", "message":"no post found"})`

refer to resource i shared on the channel. 

we need to work on your file/folder structure. but it's okay for now
modify those responses and you should be fine. keep it up!

wakaflorien commented on server/routes/router.js:
```diff
@@ -0,0 +1,65 @@
+			blog.image = req.body.image;
+		}
+		const blogs = await blog.save();
+		res.status(200).json(blogs);
```
stay consistent with the response as i mentioned.

wakaflorien commented on server/routes/router.js:
```diff
@@ -0,0 +1,65 @@
+	}
+});
+
+router.patch("/:id", async (req, res) => {
```
Preferable is `put` as primary update code name.

Shumba250 committed:
bug{messages): contact messages

debug GET API that retrieves the number of messages in the database
used JSEND response syntax

Shumba250 committed:
bug(contactMessages, blogs): fixed jsend responses

changed the syntax jsend responses

finishes(contactMesaages)
